### PR TITLE
Add back typescript-eslint/recommended

### PR DESCRIPTION
**Pull request by zakpatterson**
Most notably this catches numerous unused imports and unused declared variables.

Many other rules catch questionable code, such as use of `{}`, object, any, String, Number as types. 

Removed `import React` per new recommended rules: https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html

### CODE REVIEW

File: src/components/Summary.tsx [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import React, { Fragment, ReactElement } from 'react'
+import { Fragment, ReactElement } from 'react'
```
Comment (bluSCALE4) [inline src/components/Summary.tsx:1]: Not sure if you want to just do away with all `Fragment`'s at this point.

Comment (zakpatterson) [reply src/components/Summary.tsx:1]: Agree, will add

**Review by bluSCALE4 — APPROVED** [review]

File: src/components/ConditionallyWrap.tsx [hunk 2]
```diff
@@ -1,9 +1,7 @@
-import React, {
-  PropsWithChildren,
-  FunctionComponent,
-  ReactElement
-} from 'react'
+import { PropsWithChildren, FunctionComponent, ReactElement } from 'react'
 
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+/* eslint-disable @typescript-eslint/no-explicit-any */
```
Comment (zakpatterson) [inline src/components/ConditionallyWrap.tsx:4]: @bluSCALE4 I wasn't able to figure out a nice way to remove these disables, would you take a quick look

Comment (zakpatterson) [reply src/components/ConditionallyWrap.tsx:4]: One problem I see is that `PropsWIthChildren<{...children: ReactElement}>` is used, which duplicates the children key twice.

**Review by thegrims — APPROVED** [review]

**Pull request merged**